File: bench/rust/fibonacci.rs
use std::time::Instant;

fn fib(n: i32) -> i32 {
    if n <= 1 { return n; }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    let start = Instant::now();
    let result = fib(40);
    let ms = start.elapsed().as_secs_f64() * 1000.0;
    println!("fib(40) = {}", result);
    println!("RESULT:{:.3}", ms);
}

File: bench/js/fibonacci.js
function fib(n) {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
}

const start = process.hrtime.bigint();
const result = fib(40);
const ms = Number(process.hrtime.bigint() - start) / 1e6;
console.log(`fib(40) = ${result}`);
console.log(`RESULT:${ms.toFixed(3)}`);
